goreq.go

```diff
@@ -309,7 +315,7 @@ func (r Request) Do() (*Response, error) {
 
 	timeout := false
 	var timer *time.Timer
-	if r.Timeout > 0 {
+	if transport, ok := transport.(transportRequestCanceler); ok && r.Timeout > 0 {
```

Comment:
@nemothekid seems fair. We should stick to this so other transports can be plugged eventually.